pkg/services/provisioning/alerting/file/rules_types.go

```diff
@@ -176,88 +177,45 @@ func (queryV1 *QueryV1) mapToModel() (models.AlertQuery, error) {
 	}, nil
 }
 
-// Response structs
-
-// AlertingFileExport is the full provisioned file export.
-// swagger:model
-type AlertingFileExport struct {
-	APIVersion int64                  `json:"apiVersion" yaml:"apiVersion"`
-	Groups     []AlertRuleGroupExport `json:"groups" yaml:"groups"`
-}
-
-// AlertRuleGroupExport is the provisioned file export of AlertRuleGroupV1.
-type AlertRuleGroupExport struct {
-	OrgID    int64             `json:"orgId" yaml:"orgId"`
-	Name     string            `json:"name" yaml:"name"`
-	Folder   string            `json:"folder" yaml:"folder"`
-	Interval model.Duration    `json:"interval" yaml:"interval"`
-	Rules    []AlertRuleExport `json:"rules" yaml:"rules"`
-}
-
-// AlertRuleExport is the provisioned file export of models.AlertRule.
-type AlertRuleExport struct {
-	UID          string                     `json:"uid" yaml:"uid"`
-	Title        string                     `json:"title" yaml:"title"`
-	Condition    string                     `json:"condition" yaml:"condition"`
-	Data         []AlertQueryExport         `json:"data" yaml:"data"`
-	DashboardUID string                     `json:"dasboardUid,omitempty" yaml:"dashboardUid,omitempty"`
-	PanelID      int64                      `json:"panelId,omitempty" yaml:"panelId,omitempty"`
-	NoDataState  models.NoDataState         `json:"noDataState" yaml:"noDataState"`
-	ExecErrState models.ExecutionErrorState `json:"execErrState" yaml:"execErrState"`
-	For          model.Duration             `json:"for" yaml:"for"`
-	Annotations  map[string]string          `json:"annotations,omitempty" yaml:"annotations,omitempty"`
-	Labels       map[string]string          `json:"labels,omitempty" yaml:"labels,omitempty"`
-	IsPaused     bool                       `json:"isPaused" yaml:"isPaused"`
-}
-
-// AlertQueryExport is the provisioned export of models.AlertQuery.
-type AlertQueryExport struct {
-	RefID             string                   `json:"refId" yaml:"refId"`
-	QueryType         string                   `json:"queryType,omitempty" yaml:"queryType,omitempty"`
-	RelativeTimeRange models.RelativeTimeRange `json:"relativeTimeRange,omitempty" yaml:"relativeTimeRange,omitempty"`
-	DatasourceUID     string                   `json:"datasourceUid" yaml:"datasourceUid"`
-	Model             map[string]interface{}   `json:"model" yaml:"model"`
-}
-
 // NewAlertingFileExport creates an AlertingFileExport DTO from []AlertRuleGroupWithFolderTitle.
-func NewAlertingFileExport(groups []AlertRuleGroupWithFolderTitle) (AlertingFileExport, error) {
-	f := AlertingFileExport{APIVersion: 1}
+func NewAlertingFileExport(groups []AlertRuleGroupWithFolderTitle) (definitions.AlertingFileExport, error) {
```

Comment:
I think this converter and its helpers should also move to `ngalert/api/compat.go`, or perhaps a new file in that package, to correspond with the structs moving to `definitions`. The only caller of this is in `api/api_provisioning.go`, so this function isn't really any different to `ApiAlertRuleGroupFromAlertRuleGroup` in terms of purpose and dependencies

Reply:
I think one of the main reasons for this package to exist was to keep `AlertRuleGroupV1` and the various `Alert...Export` struct definitions close to eachother.

Since we're now separating them I think we can probably get rid of this package now, if we wanted to. Any other dependency reasons that necessitated its creation no longer exist from what I can see. It should be as simple as moving `AlertRuleGroupWithFolderTitle` to `services/ngalert/models` and then putting `services/provisioning/alerting/file/rules_types.go` + test file back into `services/provisioning/alerting`.

Reply:
moved AlertRuleGroupWithFolderTitle to ngalert/models and adapters to compat in ngalert/api

Reply:
moved files back to package where they used to be before https://github.com/grafana/grafana/pull/58623